pkg/client/client.go

```diff
@@ -0,0 +1,428 @@
+package onepassword
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging/zap/ctxzap"
+	"go.uber.org/zap"
+)
+
+type OnePasswordClient struct {
+	authType string
+	token    string
+}
+
+func NewCli(authType string, token string) *OnePasswordClient {
+	return &OnePasswordClient{
+		authType: authType,
+		token:    token,
+	}
+}
+
+func NewAccount(address string, email string, secret string, password string) *AccountDetails {
+	return &AccountDetails{
+		address:  address,
+		email:    email,
+		secret:   secret,
+		password: password,
+	}
+}
+
+// Get the accounts listed on the local config.
+func GetLocalAccounts(ctx context.Context) ([]LocalAccountDetails, error) {
+	l := ctxzap.Extract(ctx)
+
+	var err error
+	var output []byte
+
+	cmd := exec.Command("op", "accounts", "list", "--format=json")
+	if output, err = cmd.Output(); err != nil {
+		var exitErr *exec.ExitError
+		if errors.As(err, &exitErr) {
+			l.Error(
+				"error executing 'op accounts list' command",
+				zap.Error(err),
+				zap.String("stderr", string(exitErr.Stderr)),
+				zap.Int("exit_code", exitErr.ExitCode()),
+			)
+		}
+		return nil, fmt.Errorf("error executing command: %w", err)
+	}
+
+	var accounts []LocalAccountDetails
+	if err = json.Unmarshal(output, &accounts); err != nil {
+		return nil, fmt.Errorf("error unmarshalling response: %w", err)
+	}
+
+	return accounts, nil
+}
+
+// Returns the account UUID.
+func GetLocalAccountUUID(ctx context.Context, email string) (string, error) {
+	accounts, err := GetLocalAccounts(ctx)
+	if err != nil {
+		return "", fmt.Errorf("error getting local accounts: %w", err)
+	}
+
+	for _, account := range accounts {
+		if account.Email == email {
+			return account.AccountUUID, nil
+		}
+	}
+
+	return "", nil
+}
+
+// Adds a user account to the local config.
+func AddLocalAccount(ctx context.Context, providedAccountDetails *AccountDetails) (string, error) {
+	l := ctxzap.Extract(ctx)
+
+	var (
+		err     error
+		addIn   io.WriteCloser
+		account string
+	)
+
+	// Must be provided as environment variable
+	if err := os.Setenv("OP_SECRET_KEY", providedAccountDetails.secret); err != nil {
+		return "", err
+	}
+
+	args := []string{"account", "add", "--address", providedAccountDetails.address, "--email", providedAccountDetails.email, "--raw"}
```

Comment:
These values all come from the config and run on the cli. What if the the `providedAccountDetails.addres` is some variant of `foo@email.com && rm -rf / && ...`